### PR TITLE
feat(form): Converted form component to typescript

### PR DESCRIPTION
**Pull request by jessiehuff**
Fixes issue #1992

### CODE REVIEW

**Comment by nicolethoen** [comment]
@jessiehuff can you also add integration tests for this component as outlined on this [README](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-4/react-integration)?

File: packages/patternfly-4/react-core/src/components/Form/ActionGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export interface ActionGroupProps extends React.HTMLProps<HTMLDivElement> {
+  children?: React.ReactNode;
+  className?: string;
+}
+
+export const ActionGroup: React.FunctionComponent<ActionGroupProps> = ({
+  children = null,
+  className = '',
+  ...props
+}) => {
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Form/ActionGroup.tsx]: Change to `}: ActionGroupProps) => {`

File: packages/patternfly-4/react-core/src/components/Form/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  className?: string;
+  isHorizontal?: boolean;
+}
+
+export const Form: React.FunctionComponent<FormProps> = ({
+  children = null,
+  className = '',
+  isHorizontal = false,
+  ...props
+}) => (
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Form/Form.tsx]: Change to `}: FormProps) => (`.

File: packages/patternfly-4/react-core/src/components/Form/FormHelperText.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  className?: string;
+}
+
+export const FormHelperText: React.FunctionComponent<FormHelperTextProps> = ({
+  children = null,
+  isError = false,
+  isHidden = true,
+  className = '',
+  ...props
+}) => (
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Form/FormHelperText.tsx]: Change to `}: FormHelperTextProps) => (`

File: packages/patternfly-4/react-core/src/components/Form/__snapshots__/FormGroup.test.tsx.snap [hunk 4]
```diff
@@ -1,22 +1,20 @@
...
+  fieldId="id"
   label={
     <div>
       label
     </div>
   }
 >
   <div
-    className="pf-c-form__group"
+    className="pf-c-form__group pf-m-inline"
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Form/__snapshots__/FormGroup.test.tsx.snap]: why did this change?

File: packages/patternfly-4/react-core/src/components/Form/__snapshots__/FormHelperText.test.tsx.snap [hunk 5]
```diff
@@ -2,16 +2,16 @@
 
 exports[`LoginFooterItem  with custom node 1`] = `
 <p
-  className="pf-c-form__helper-text pf-m-hidden"
+  className="pf-c-form__helper-text pf-m-error"
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Form/__snapshots__/FormHelperText.test.tsx.snap]: Why did this change?

File: packages/patternfly-4/react-integration/cypress/integration/form.spec.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+describe('Form Demo Test', () => {
+  it('Navigate to demo section', () => {
+    cy.visit('http://localhost:3000/');
+    cy.get('#form-demo-nav-item-link').click();
```
Comment (tlabaj) [inline packages/patternfly-4/react-integration/cypress/integration/form.spec.ts]: Can you please add some interactive test with the form.

**Pull request merged**